mailserver/mailserver.go

```diff
@@ -188,6 +197,15 @@ func (s *WMailServer) DeliverMail(peer *whisper.Peer, request *whisper.Envelope)
 		return
 	}
 
+	// Recover from possible goleveldb panics
+	defer func() {
+		if r := recover(); r != nil {
+			if errString, ok := r.(string); ok {
+				log.Error(fmt.Sprintf("recovered from panic in DeliverMail: %s", errString))
+			}
+		}
+	}()
```

Comment:
I thought this part should have been in `processRequest` but I think it makes sense have it here so that `sendHistoricMessageResponse` is not called if `processRequest` panics.